app/src/main/java/org/mifos/mobile/ui/fragments/RegistrationFragment.kt

```diff
@@ -41,195 +43,207 @@ class RegistrationFragment : BaseFragment(), RegistrationView {
+                override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
+                    if (charSequence.isEmpty()) {
+                        progressBar.visibility = View.GONE
+                        passwordStrength.visibility = View.GONE
+                    } else {
+                        progressBar.visibility = View.VISIBLE
+                        passwordStrength.visibility = View.VISIBLE
+                        updatePasswordStrengthView(charSequence.toString())
+                    }
                 }
-            }
-
-            override fun afterTextChanged(editable: Editable) {}
-        })
+                override fun afterTextChanged(editable: Editable) {}
+            })
+        }
         return rootView
     }
 
     private fun updatePasswordStrengthView(password: String) {
-        if (TextView.VISIBLE != binding.passwordStrength.visibility) return
-        if (password.isEmpty()) {
-            binding.passwordStrength.text = ""
-            binding.progressBar.progress = 0
-            return
-        }
-        val str = PasswordStrength.calculateStrength(password)
-        binding.passwordStrength.text = str.getText(context)
-        binding.passwordStrength.setTextColor(str.color)
-        val mode = PorterDuff.Mode.SRC_IN
-        binding.progressBar.progressDrawable?.setColorFilter(str.color, mode)
-        if (str.getText(context) == getString(R.string.password_strength_weak)) {
-            binding.progressBar.progress = 25
-        } else if (str.getText(context) == getString(R.string.password_strength_medium)) {
-            binding.progressBar.progress = 50
-        } else if (str.getText(context) == getString(R.string.password_strength_strong)) {
-            binding.progressBar.progress = 75
-        } else {
-            binding.progressBar.progress = 100
+        with(binding) {
+            if (TextView.VISIBLE != passwordStrength.visibility) return
+            if (password.isEmpty()) {
+                passwordStrength.text = ""
+                progressBar.progress = 0
+                return
+            }
+            val str = PasswordStrength.calculateStrength(password)
+            passwordStrength.text = str.getText(context)
+            passwordStrength.setTextColor(str.color)
+            val mode = PorterDuff.Mode.SRC_IN
+            progressBar.progressDrawable?.setColorFilter(str.color, mode)
+            if (str.getText(context) == getString(R.string.password_strength_weak)) {
+                progressBar.progress = 25
+            } else if (str.getText(context) == getString(R.string.password_strength_medium)) {
+                progressBar.progress = 50
+            } else if (str.getText(context) == getString(R.string.password_strength_strong)) {
+                progressBar.progress = 75
+            } else {
+                progressBar.progress = 100
+            }
```

Comment:
This section can be improved a lot but for now, at least can refactore this to use Kotlin's `when`